### PR TITLE
[certificate-acl] Added an http endpoint to get an acl entry

### PR DESCRIPTION
**Pull request by rascaraficci**
* **Please check if the PR fulfills these requirements**
- [ X ] Tests for the changes have been added (for bug fixes / features)
- [ X ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
There is no endpoint to get an acl entry. The only way is to access the database.

* **What is the new behavior (if this is a feature change)?**
It was added an http endpoint to get an acl entry.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:
dojot/dojot#1939

### CODE REVIEW

File: iam/certificate-acl/README.md [hunk 1]
```diff
@@ -1 +1,108 @@
...
+|-----|------|---------------|--------------|----------------------|---------|
+| enable.async.commit| boolean | `true` | `true` or `false` | CERTIFICATE_ACL_ENABLE_ASYNC_COMMIT | Enable asynchronous processing mode. |
+| commit.on.failure| boolean | `false` | `true` or `false` | CERTIFICATE_ACL_ENABLE_ASYNC_COMMIT | Enable to commit even if the processing callback has failed. |
+| kafka.topic.auto.offset.reset | string | `earliest` | `latest`, `earliest`, `none` | CERTIFICATE_ACL_KAFKA_TOPIC_AUTO_OFFSET_RESET | What to do when there is no initial offset in Kafka or if the current offset does not exist any more on the server. See more [here](https://kafka.apache.org/documentation/#consumerconfigs_auto.offset.reset). |
+| kafka.consumer.group.id | string | `certificate-acl` | | CERTIFICATE_ACL_KAFKA_CONSUMER_GROUP_ID | A unique string that identifies the consumer group this consumer belongs to. See more [here](https://kafka.apache.org/documentation/#consumerconfigs_group.id). |
+| kafka.consumer.metadata.broker.list | string | `kafka:9092` | - | CERTIFICATE_ACL_KAFKA_CONSUMER_METADATA_BROKER_LIST | A comma-separated list containing the kafka's `host:port`. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:89]: CERTIFICATE_ACL_CONSUMER_METADATA_BROKER_LIST

Comment (mprevide) [inline iam/certificate-acl/README.md:89]: consumer.metadata.broker.list

File: iam/certificate-acl/config/default.conf [hunk 2]
```diff
@@ -1,33 +1,44 @@
...
 # LOGGER
-log.verbose:boolean=false
-log.console.level=info
-log.file.enable:boolean=false
-log.file.level=info
-log.file.filename=certificate-acl-logs-%DATE%.log
+logger.verbose:boolean=false
+logger.console.level=info
+logger.file.enable:boolean=false
+logger.file.leve:string=info
```
Comment (mprevide) [inline iam/certificate-acl/config/default.conf:9]: logger.file.level

File: iam/certificate-acl/README.md [hunk 3]
```diff
@@ -1 +1,108 @@
...
+
+#### Logger settings
+
+| Key | type | Default Value | Valid Values | Environment variable | Purpose |
+|-----|------|---------------|--------------|----------------------|---------|
+| logger.verbose | boolean | `false` | `true` or `false` | CERTIFICATE_ACL_LOGGER_VERBOSE | In verbose mode, file and line of where the logging method has been called are added as metadata to the logging message. |
+| logger.console.level | string | `info` | `error`, `warn`, `info` or `debug` | CERTIFICATE_ACL_LOGGER_CONSOLE_LEVEL | logging level. |
+| logger.file.enable | boolean | `false` | `true` or `false` | CERTIFICATE_ACL_LOGGER_FILE_ENABLE | Flag that enables logging to file. |
+| logger.file.level | string | `info` | `error`, `warn`, `info` or `debug` | CERTIFICATE_ACL_LOGGER_FILE_LEVEL | logging level. |
+| logger.file.dir | string | `./temp/log/` | | CERTIFICATE_ACL_LOGGER_FILE_DIR | Directory where the log files will be saved. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:76]: Is this really used?

File: iam/certificate-acl/README.md [hunk 4]
```diff
@@ -1 +1,108 @@
...
+| logger.file.name | string | `dojot.certificate-acl-%DATE%.log` | | CERTIFICATE_ACL_LOGGER_FILE_NAME | Log file name pattern. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:77]: file.filename?

File: iam/certificate-acl/README.md [hunk 5]
```diff
@@ -1 +1,108 @@
...
+| logger.file.max | string | `7d` | | CERTIFICATE_ACL_LOGGER_FILE_MAX | Maximum number of logs to keep. If not set, no logs will be removed. This can be a number of files or number of days. If using days, add 'd' as the suffix. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:78]: Is this really used?

File: iam/certificate-acl/README.md [hunk 6]
```diff
@@ -1 +1,108 @@
...
+| logger.file.size | string | `10m` | | CERTIFICATE_ACL_LOGGER_FILE_SIZE | Maximum size of the file after which it will rotate. This can be a number of bytes, or units of kb, mb, and gb. If using the units, add 'k', 'm', or 'g' as the suffix. The units need to directly follow the number. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:79]: Is this really used?

File: iam/certificate-acl/README.md [hunk 7]
```diff
@@ -1 +1,108 @@
...
+
+#### Kafka integration settings
+
+| Key | type | Default Value | Valid Values | Environment variable | Purpose |
+|-----|------|---------------|--------------|----------------------|---------|
+| enable.async.commit| boolean | `true` | `true` or `false` | CERTIFICATE_ACL_ENABLE_ASYNC_COMMIT | Enable asynchronous processing mode. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:85]: kafka.enable.async.commit

Comment (mprevide) [inline iam/certificate-acl/README.md:85]: CERTIFICATE_ACL_KAFKA_ENABLE_ASYNC_COMMIT

File: iam/certificate-acl/README.md [hunk 8]
```diff
@@ -1 +1,108 @@
...
+| commit.on.failure| boolean | `false` | `true` or `false` | CERTIFICATE_ACL_ENABLE_ASYNC_COMMIT | Enable to commit even if the processing callback has failed. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:86]: kafka.on.failure

Comment (mprevide) [reply iam/certificate-acl/README.md:86]: CERTIFICATE_ACL_KAFKA_ENABLE_ASYNC_COMMIT

File: iam/certificate-acl/README.md [hunk 9]
```diff
@@ -1 +1,108 @@
...
+| kafka.topic.auto.offset.reset | string | `earliest` | `latest`, `earliest`, `none` | CERTIFICATE_ACL_KAFKA_TOPIC_AUTO_OFFSET_RESET | What to do when there is no initial offset in Kafka or if the current offset does not exist any more on the server. See more [here](https://kafka.apache.org/documentation/#consumerconfigs_auto.offset.reset). |
```
[the remark this answers is not in the document]
Comment (mprevide) [reply iam/certificate-acl/README.md:87]: CERTIFICATE_ACL_TOPIC_AUTO_OFFSET_RESET

File: iam/certificate-acl/README.md [hunk 10]
```diff
@@ -1 +1,108 @@
...
+| kafka.consumer.group.id | string | `certificate-acl` | | CERTIFICATE_ACL_KAFKA_CONSUMER_GROUP_ID | A unique string that identifies the consumer group this consumer belongs to. See more [here](https://kafka.apache.org/documentation/#consumerconfigs_group.id). |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:88]: consumer.group.id

File: iam/certificate-acl/README.md (hunk 1 above)
Comment (mprevide) [inline iam/certificate-acl/README.md:89]: consumer.metadata.broker

File: iam/certificate-acl/README.md [hunk 11]
```diff
@@ -1 +1,108 @@
...
+|redis.port | integer | 6379 | | CERTIFICATE_ACL_REDIS_PORT | Port for the Redis database. |
+|redis.db | integer | 0 | [0-15] | CERTIFICATE_ACL_REDIS_DB | Redis database to be used. |
+|redis.reconnect.after.ms | integer | `5000` | `[0,...]` | CERTIFICATE_ACL_REDIS_RECONNECT_AFTER_MS | Time in milliseconds to wait before reconnecting to the redis database.  |
+| redis.operation.timeout.ms | integer | `1000` | `[0,...]` | CERTIFICATE_ACL_REDIS_OPERATION_TIMEOUT_MS | Timeout in milliseconds for an operation in the redis database. |
+
+#### X509-identity-mgmt integration settings
+
+| Key | type | Default Value | Valid Values | Environment variable | Purpose |
+|-----|------|---------------|--------------|----------------------|---------|
+|service.hostname | string | `x509-identity-mgmt` | | CERTIFICATE_ACL_SERVICE_HOSTNAME | Hostname for the x509-identity-mgmt service. |
```
Comment (mprevide) [inline iam/certificate-acl/README.md:105]: Chante to x509.hostname ?

File: iam/certificate-acl/app/redis/RedisManager.js [hunk 12]
```diff
@@ -27,28 +33,33 @@ class RedisManager {
    * Creates a Redis Manager
    *
    * @constructor
+   * @param {*} serviceStateManager instance of a ServiceStateManager
    * @returns Redis Manager Object
    */
-  constructor() {
+  constructor(serviceStateManager) {
```
Comment (mprevide) [inline iam/certificate-acl/app/redis/RedisManager.js:39]: I'm not sure if it's really necessary, but remember that issue that we thought was a lib problem? #2094

https://github.com/dojot/dojot/blob/46aef2686aa4d4c1c7dacd2c16c84416597d5484/subscription-engine/kafka-ws/app/Redis/Utils.js#L15
https://github.com/dojot/dojot/blob/46aef2686aa4d4c1c7dacd2c16c84416597d5484/subscription-engine/kafka-ws/app/Redis/RedisManager.js#L73

File: iam/certificate-acl/app/server/createFramework.js [hunk 13]
```diff
@@ -0,0 +1,14 @@
+const {
+  WebUtils, Logger,
+} = require('@dojot/microservice-sdk');
+
+const logger = new Logger('certificate-acl:express');
+
+module.exports = (aclRoute) => WebUtils.framework.createExpress(
+  {
```
Comment (mprevide) [inline iam/certificate-acl/app/server/createFramework.js:8]: Perhaps some of these middleware are interesting, with readinessInterceptor and beaconInterceptor

https://github.com/dojot/backstage/blob/ca9ffa2a205e7c09265cd28e2458bedc7f6d3580/app/express/index.js#L38

**Review by mprevide — CHANGES_REQUESTED** [review]
I left some comments

File: iam/certificate-acl/config/default.conf (hunk 2 above)
Comment (rascaraficci) [reply iam/certificate-acl/config/default.conf:9]: :+1:

File: iam/certificate-acl/README.md (hunk 4 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md:77]: Your totally right! I copied from https://github.com/dojot/dojot/tree/development/iam/x509-identity-mgmt#logger-settings

File: iam/certificate-acl/README.md (hunk 5 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md:78]: Removed!

File: iam/certificate-acl/README.md (hunk 6 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md:79]: Removed!

File: iam/certificate-acl/README.md (hunk 3 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md:76]: Removed!

File: iam/certificate-acl/README.md (hunk 7 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md:85]: :+1:

Comment (rascaraficci) [reply iam/certificate-acl/README.md:85]: :+1:

File: iam/certificate-acl/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (rascaraficci) [reply iam/certificate-acl/README.md:87]: :+1:

File: iam/certificate-acl/README.md (hunk 10 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md:88]: :+1:

File: iam/certificate-acl/README.md (hunk 1 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md:89]: :+1:

Comment (rascaraficci) [reply iam/certificate-acl/README.md:89]: :+1:

File: iam/certificate-acl/README.md [hunk 14]
```diff
@@ -1 +1,108 @@
...
+|service.port | integer | `3000` |  | CERTIFICATE_ACL_SERVICE_PORT | Port for the x509-identity-mgmt service. |
```
[the remark this answers is not in the document]
Comment (rascaraficci) [reply iam/certificate-acl/README.md:106]: Just x509 seems you're saying about a certificate. I'll change to x509im.

**Review by cfrancisco — DISMISSED** [review]

File: iam/certificate-acl/README.md (hunk 11 above)
Comment (rascaraficci) [reply iam/certificate-acl/README.md]: I changed `service` by `x509im`.

File: iam/certificate-acl/app/server/createFramework.js (hunk 13 above)
Comment (rascaraficci) [reply iam/certificate-acl/app/server/createFramework.js:16]: I'm not so familiar with the webutils. Thanks for point out this!

File: iam/certificate-acl/app/redis/RedisManager.js (hunk 12 above)
Comment (mprevide) [reply iam/certificate-acl/app/redis/RedisManager.js:39]: put this how TODO and I will approve the code

Comment (rascaraficci) [reply iam/certificate-acl/app/redis/RedisManager.js:39]: :+1:

**Review by mprevide — APPROVED** [review]

**Pull request merged**